### PR TITLE
Audio inflator

### PR DESCRIPTION
**Pull request by uhunkler**
I have created the audio and audio-params inflators and added a background sound and a positional sound emitted by a speaker to the playground scene. I have also added the sound files for the outdoor-festival scene and updated the index.html to handle the scene audio. The third-party assets/sounds licenses I added in a licenses.md file

Please check if everything works for you.

### CODE REVIEW

**Comment by vincentfretin** [comment]
You shouldn't need to include the outdoor festival glb and sounds in this repo.
The code I provided in https://github.com/c-frame/aframe-gltf-model-plus/issues/9#issuecomment-2039493171 didn't work?

**Comment by uhunkler** [comment]
I misunderstood your code I provided in https://github.com/c-frame/aframe-gltf-model-plus/issues/9#issuecomment-2039493171. I thought it is just an example, I'll change the code in index.html, remove the files and test the changes ...

File: examples/playground/index.html [hunk 1]
```diff
@@ -127,6 +143,7 @@ <h2 style="margin: 0; font-size: 1.2rem">Help</h2>
     </script>
 
     <a-scene
+      audio
```
Comment (vincentfretin) [inline examples/playground/index.html:146]: Actually you don't really need to add it if you don't set any option, and there is none. All systems are always included in the scene.

File: examples/playground/index.html [hunk 2]
```diff
@@ -160,7 +177,7 @@ <h2 style="margin: 0; font-size: 1.2rem">Help</h2>
         <a-entity id="player" class="camera" camera position="0 1.6 0" look-controls></a-entity>
       </a-entity>
 
-      <a-entity class="environment-settings" gltf-model-plus="#sceneGLB"></a-entity>
+      <a-entity id="gltf-model-plus-entity" class="environment-settings" gltf-model-plus="#sceneGLB"></a-entity>
```
Comment (vincentfretin) [inline examples/playground/index.html:180]: You added this id but it doesn't seem to be used.

File: src/inflators/utils.js [hunk 3]
```diff
@@ -30,3 +30,9 @@ export const createEntityAndReparent = (node, klass = undefined) => {
   sceneEl.object3D.attach(node);
   return entity;
 };
+
+export const absoluteURLForAsset =
+  window.absoluteURLForAsset ||
+  ((asset) => {
+    return asset;
+  });
```
Comment (vincentfretin) [inline src/inflators/utils.js:38]: Can you please remove it from inflators/environment-settings.js and import it from here as well.

File: src/inflators/audio.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  const audioProps = {
+    autoplay: componentProps.autoPlay,
+    distanceModel: audioParams.distanceModel,
+    loop: componentProps.loop,
+    maxDistance: audioParams.maxDistance,
+    positional: audioParams.audioType !== "stereo",
+    refDistance: audioParams.refDistance,
+    rolloffFactor: audioParams.rolloffFactor,
+    src: absoluteURLForAsset(componentProps.src),
+    volume: hasAudioParams ? audioParams.gain : audioParams.volume * 3,
```
Comment (vincentfretin) [inline src/inflators/audio.js:20]: Multiplying by 3 is what hubs code does?

File: examples/playground/index.html [hunk 5]
```diff
@@ -10,12 +10,26 @@
...
     <script src="https://cdn.jsdelivr.net/npm/aframe-look-at-component@1.0.0/dist/aframe-look-at-component.min.js"></script>
     <script src="https://cdn.jsdelivr.net/gh/networked-aframe/naf-valid-avatars@4b116a4/src/player-info.js"></script>
+    <script>
+      // @var url {string} The url as string.
+      window.absoluteURLForAsset = (url) => {
+        let newurl = "";
+
+        if (!url) return "";
+        if (url.startsWith("https://vr.unodo-design.com/assets/")) {
+          newurl = url.replace("https://vr.unodo-design.com/assets/", "http://localhost:8080/examples/playground/");
```
[the remark this answers is not in the document]
Comment (uhunkler) [reply examples/playground/index.html:27]: I should have replaced localhost with ./ what I'll do now.

File: src/inflators/audio.js (hunk 4 above)
Comment (uhunkler) [reply src/inflators/audio.js:20]: No, but without the multiply the sound is so low, that you can hardly hear it. I didn't find what values are used for the alternative audio component, so I tried to get about the same volume I hear when the GLB is loaded in Hubs.

File: examples/playground/index.html (hunk 2 above)
Comment (uhunkler) [reply examples/playground/index.html:180]: Right, that's a leftover from my first approach.

File: src/inflators/audio.js (hunk 4 above)
Comment (vincentfretin) [reply src/inflators/audio.js:20]: Ok fair we don't have the same exact sound system/component as hubs here, so handling the volume may be different. And if we have an audio-params component, there is not need to multiply by 3?

File: src/inflators/utils.js (hunk 3 above)
Comment (uhunkler) [reply src/inflators/utils.js:38]: Done.

**Comment by uhunkler** [comment]
All changes you listed I have implemented and the changed code is committed and pushed. You may check again. 

I'll leave your comments unresolved for now because it is easier to follow up when they stay unresolved for now.

**Comment by vincentfretin** [comment]
You can remove Outdoor_Festival.glb from the repo too.

**Comment by vincentfretin** [comment]
I see hubs supports setting panningModel coneInnerAngle coneOuterAngle coneOuterGain for positional audio.
https://github.com/mozilla/hubs/blob/746413d43cba0a00f79f66b77fc9e559230c64ba/src/update-audio-settings.js#L26-L29
and aframe sound component does not. Later, we may contribute a PR to aframe to support those if it really makes a difference. I'm not familiar with exactly what those params do.

**Review by vincentfretin — APPROVED** [review]
Ok let's merge this. Thank you very much!

**Comment by vincentfretin** [comment]
Actually in hubs coneInnerAngle coneOuterAngle coneOuterGain may not be used
https://github.com/mozilla/hubs/blob/746413d43cba0a00f79f66b77fc9e559230c64ba/src/systems/audio-gain-system.js#L32
I'm not sure which code to look at in the hubs repo, too complex :D

**Comment by uhunkler** [comment]
> I see hubs supports setting panningModel coneInnerAngle coneOuterAngle coneOuterGain for positional audio. https://github.com/mozilla/hubs/blob/746413d43cba0a00f79f66b77fc9e559230c64ba/src/update-audio-settings.js#L26-L29 and aframe sound component does not. Later, we may contribute a PR to aframe to support those if it really makes a difference. I'm not familiar with exactly what those params do.

I never tired those values - I guess you can set some kind of directional audio.

**Comment by uhunkler** [comment]
@vincentfretin are you going to build the dist with the changes?

**Comment by vincentfretin** [comment]
yes, done, otherwise Github Pages doesn't play the sounds. ;)

**Pull request merged**